src/main/java/uk/gov/pay/api/auth/AccountAuthenticator.java

```diff
@@ -41,8 +42,19 @@ public AccountAuthenticator(Client client, PublicApiConfig configuration) {
                 .get();
 
         if (response.getStatus() == OK.getStatusCode()) {
-            return readAccountFromResponse(response);
+            AuthResponse authResponse = response.readEntity(AuthResponse.class);
```

Comment:
Organising this into a simple class is great 👍 . It looks like the behaviour to assign the default `CARD` token type if there is none present is removed here, just sense checking that that will play nice with serialisation/ deserialisation here and upstream (public auth).